RELEASE.md

```diff
@@ -3,17 +3,19 @@
 The Kubespray Project is released on an as-needed basis. The process is as follows:
 
 1. An issue is proposing a new release with a changelog since the last release. Please see [a good sample issue](https://github.com/kubernetes-sigs/kubespray/issues/8325)
-2. At least one of the [approvers](OWNERS_ALIASES) must approve this release
-3. The `kube_version_min_required` variable is set to `n-1`
-4. Remove hashes for [EOL versions](https://github.com/kubernetes/website/blob/main/content/en/releases/patch-releases.md) of kubernetes from `*_checksums` variables.
-5. Create the release note with [Kubernetes Release Notes Generator](https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md). See the following `Release note creation` section for the details.
-6. An approver creates [new release in GitHub](https://github.com/kubernetes-sigs/kubespray/releases/new) using a version and tag name like `vX.Y.Z` and attaching the release notes
-7. An approver creates a release branch in the form `release-X.Y`
-8. The corresponding version of [quay.io/kubespray/kubespray:vX.Y.Z](https://quay.io/repository/kubespray/kubespray) and [quay.io/kubespray/vagrant:vX.Y.Z](https://quay.io/repository/kubespray/vagrant) container images are built and tagged. See the following `Container image creation` section for the details.
-9. The `KUBESPRAY_VERSION` variable is updated in `.gitlab-ci.yml`
-10. The release issue is closed
-11. An announcement email is sent to `dev@kubernetes.io` with the subject `[ANNOUNCE] Kubespray $VERSION is released`
-12. The topic of the #kubespray channel is updated with `vX.Y.Z is released! | ...`
+1. At least one of the [approvers](OWNERS_ALIASES) must approve this release
+1. (Only for major releases) The `kube_version_min_required` variable is set to `n-1`
+1. (Only for major releases) Remove hashes for [EOL versions](https://github.com/kubernetes/website/blob/main/content/en/releases/patch-releases.md) of kubernetes from `*_checksums` variables.
+1. Create the release note with [Kubernetes Release Notes Generator](https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md). See the following `Release note creation` section for the details.
+1. An approver creates [new release in GitHub](https://github.com/kubernetes-sigs/kubespray/releases/new) using a version and tag name like `vX.Y.Z` and attaching the release notes
+1. (Only for major releases) An approver creates a release branch in the form `release-X.Y`
+1. (Only for major releases) On the `master` branch: bump the version in `galaxy.yml` to the next expected major release (X.y.0 with y = Y + 1), commit and push.
```

Comment:
Well, I was reusing the Ansible guidelines ; the idea being that as soon as we cut a new release, the very next thing we do is increment the version number (aka galaxy.yml version should match KUBESPRAY_NEXT_VERSION).
I think that way it would be harder to release with the wrong version, since it's checked by CI during all the time.

Reply:
I don't have a strong opinion on that part. The one downside I can see for that is that it kinda breaks the workflow of doing a release (since you have to make a PR and get a lgtm for what is a really trivial change).
It's not **that** bad ofc, so I can change it if you like

----
Ideally we would have automation for that (prow creating a PR for updating the version number whenever it sees a releases), but well, we don't have that currently ^.

Reply:
Also if we go with that we'd have to use KUBESPRAY_NEXT_VERSION instead of KUBEPRAY_VERSION in the check-galaxy script

Reply:
The idea of automation is great, though we need to do it manually.👍
And the PR is better than a push because it can make the process more transparent :-)


> I don't have a strong opinion on that part. The one downside I can see for that is that it kinda breaks the workflow of doing a release (since you have to make a PR and get a lgtm for what is a really trivial change). It's not **that** bad ofc, so I can change it if you like
> 
> Ideally we would have automation for that (prow creating a PR for updating the version number whenever it sees a releases), but well, we don't have that currently ^.

Reply:
> Also, if we go with that, we'd have to use KUBESPRAY_NEXT_VERSION instead of KUBEPRAY_VERSION in the check-galaxy script

I agree with you; the CI code needs to be changed :-)
BTW, does the `KUBESPRAY_NEXT_VERSION, `KUBESPRAY_VERSION,` and `galaxy.yml` be changed together better?

Reply:
Yeah good point I'll add that to the docs as well :+1:

Reply:
Hum
I'm trying to refactor the check-galaxy version to have it check the next expected version, unfortunately it's not that easy since we don't have access to the target MR branch (so can't distinguish between minor/major....) :thinking: